compiler-core/src/language_server/feedback.rs

```diff
@@ -123,15 +123,29 @@ impl FeedbackBookKeeper {
         self.unset_errors(&mut feedback);
 
         for diagnostic in diagnostics {
-            match diagnostic.location.as_ref().map(|l| l.path.clone()) {
-                Some(path) => {
-                    _ = self.files_with_errors.insert(path.clone());
-                    feedback.append_diagnostic(path, diagnostic);
-                }
-
-                None => {
-                    feedback.append_message(diagnostic);
+            if let Some(l) = &diagnostic.location {
+                _ = self.files_with_errors.insert(l.path.clone());
+                feedback.append_diagnostic(l.path.clone(), diagnostic.clone());
+
+                for extra in &l.extra_labels {
```

Comment:
This renders the additional information as entirely new diagnostics rather than being part of the first one. This may be OK for imports cycles, but in the duplicate import example there is now a diagnostic that just says "First imported here" with no other context.

It should instead be a nested diagnostic, possibly of a different colour like how hints are displayed.

Reply:
Yeah that makes sense. The reason I had it like this was that it looks like sub-diagnostics can't be across files in the lsp protocol if I am reading correctly. Lemme change this so the ones in different files are separate diagnostics but the ones in the same file are sub diagnostics

Reply:
Oh nevermind I'm totally wrong and misread the protocol 🤦🏽 lemme fix this